sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -377,8 +377,7 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
    * @param newArgs the new product arguments.
    */
   def makeCopy(newArgs: Array[AnyRef]): BaseType = attachTree(this, "makeCopy") {
-    // Skip no-arg constructors that are just there for kryo.
-    val ctors = getClass.getConstructors.filter(_.getParameterTypes.size != 0)
+    val ctors = getClass.getConstructors
```

Comment:
Why this change?

Reply:
As now case class (ResetCommand) has a zero parameter constructor, we cannot filter it out

Reply:
Simply removing this does not sound right to me. We need to double check the impact. cc @cloud-fan

Reply:
See https://github.com/apache/spark/pull/5157/files

Reply:
@gatorsmile i have changed the fix to override makeCopy in ResetCommand instead now